test/standalone/dirname/build.zig

```diff
@@ -0,0 +1,72 @@
+const std = @import("std");
+
+pub fn build(b: *std.Build) void {
+    const test_step = b.step("test", "Test it");
+    b.default_step = test_step;
+
+    const touch_src = std.Build.LazyPath{
+        .path = "touch.zig",
+    };
+
+    const touch = b.addExecutable(.{
+        .name = "touch",
+        .root_source_file = touch_src,
+        .optimize = .Debug,
+    });
+    const generated = b.addRunArtifact(touch).addOutputFileArg("subdir" ++ std.fs.path.sep_str ++ "generated.txt");
+
+    const exists_in = b.addExecutable(.{
+        .name = "exists_in",
+        .root_source_file = .{ .path = "exists_in.zig" },
+        .optimize = .Debug,
+    });
+
+    const has_basename = b.addExecutable(.{
+        .name = "has_basename",
+        .root_source_file = .{ .path = "has_basename.zig" },
+        .optimize = .Debug,
+    });
+
+    // Known path:
+    addTestRun(test_step, exists_in, touch_src.dirname(), &.{"touch.zig"});
+
+    // Generated file:
+    addTestRun(test_step, exists_in, generated.dirname(), &.{"generated.txt"});
+
+    // Generated file multiple levels:
+    addTestRun(test_step, exists_in, generated.dirname().dirname(), &.{
+        "subdir" ++ std.fs.path.sep_str ++ "generated.txt",
+    });
+
+    // Cache root:
+    addTestRun(test_step, has_basename, generated.dirname().dirname().dirname().dirname(), &.{"zig-cache"});
```

Comment:
This should be the basename from b.cache_root.
I'll see if I can repro this failure locally and update this.

Reply:
Done.